app/src/lib/menu-update.ts

```diff
@@ -155,8 +155,15 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.setEnabled('pull', hasPublishedBranch && !networkActionInProgress)
     menuStateBuilder.setEnabled('create-branch', !tipStateIsUnknown)
   } else {
-    for (const id of repositoryScopedIDs) {
-      menuStateBuilder.disable(id)
+    if (selectedState && selectedState.type === SelectionType.MissingRepository) {
+      menuStateBuilder.enable('view-repository-on-github')
+      menuStateBuilder.enable('remove-repository')
+    } else {
```

Comment:
but I completely agree that is me taking advantage of the way it executes and should be fixed